db/migrations.go

```diff
@@ -8,22 +8,34 @@ import (
 // If this runs, it means the database is empty. No migrations will be applied on top of it, as this sets up the database from scratch so it starts out "up to date"
 func BaselineSchema(tx *gorm.DB) error {
 	log.Debugf("first run: initializing database schema")
-	err := tx.AutoMigrate(&Tenant{}, &Policy{}, &Address{}, &Clause{}, &Collection{}, &Label{}, &StorageProvider{}, &TenantStorageProvider{}, ReplicationConstraint{}, ConstraintAttribute{})
 
+	// Enums
+	err := tx.Exec("CREATE TYPE tenant_sp_state AS ENUM ('eligible', 'pending', 'active', 'suspended');").Error
 	if err != nil {
-		log.Fatalf("error initializing database: %s", err)
+		log.Fatalf("error creating enum: %s", err)
 	}
+
+	err = tx.Exec("CREATE TYPE comparison_operator AS ENUM ('>', '<', '=', '>=', '<=', 'in', 'nin', '!=');").Error
+	if err != nil {
+		log.Fatalf("error creating enum: %s", err)
+	}
```

Comment:
This is the only way I could make the enums work - there doesn't seem to be a way to do this directly in the Gorm schema. Seems a bit messy to decouple them this way

Reply:
These don't show up in the `schema.sql` either, but I confirmed that they get applied in a live Postgres db

Reply:
Can you not move this to an "ad-hoc" migration file? IIRC gorm does support such `free-form-text.sql` files...?

Reply:
It should be possible but we have to read them as files and execute their contents like this :

```go
sqlBytes, err := os.ReadFile("path/to/migration.sql")
            if err != nil {
                return fmt.Errorf("error reading migration.sql: %w", err)
            }

            // Execute the SQL commands
            return tx.Exec(string(sqlBytes)).Error
```

I couldn't find any "native" way to just reference sql files from within gorm, but it may make things a bit cleaner if we use Atlas (https://atlasgo.io/guides/orms/gorm)  - it seems to support versioned migrations in raw .sql files. I could explore this as an option, instead of gormigrate?

Reply:
Nod, do what makes most sense from your perspective, given balance between ick-factor and time constraints :)